### PR TITLE
[FIX] website_sale: use separate contact for pick-up point

### PR DESCRIPTION
**Pull request by khah-odoo**
Steps to reproduce:
1. Configure Sendcloud shipping with pick-up locations
2. Go to website and use the shipping method and select a pick-up location
3. Try step 2 again, using the same pick-up point but with a different name
4. Checking the delivery address of the second customer, we see the name of the first customer is used

The problem is that if a pick-up location is already saved, we re-use the same contact for the delivery address. This commit ensures separate contacts are created for different customers.

opw-3853716

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by Dotni — APPROVED** [review]
Hello @khah-odoo :wave:

The code looks good to me, well done!

@amoyaux can you take a look as well?

**Review by vchu-odoo — CHANGES_REQUESTED** [review]
Hello,

Thank you for your work!

Here is one suggestion for improvement, wdyt?

File: addons/website_sale/models/sale_order.py [hunk 1]
```diff
@@ -499,12 +499,13 @@ def _action_confirm(self):
...
             # we can check if the current partner has a partner of type "delivery" that has the same address
-            existing_partner = order.env['res.partner'].search(['&', '&', '&', '&',
+            existing_partner = order.env['res.partner'].search(['&', '&', '&', '&', '&',
                                                                 ('street', '=', street),
                                                                 ('city', '=', city),
                                                                 ('state_id', '=', state),
                                                                 ('country_id', '=', country),
-                                                                ('type', '=', 'delivery')], limit=1)
+                                                                ('type', '=', 'delivery'),
+                                                                ('parent_id', '=', order.partner_id.id)], limit=1)
```
[the remark this answers is not in the document]
Comment (Dotni) [reply addons/website_sale/models/sale_order.py:508]: Good catch!

**Review by vchu-odoo — APPROVED** [review]
LGTM :+1:

**Review by Feyensv — COMMENTED** [review]
Is the bug only there from 17.0 ? If so, why?

**Comment by vchu-odoo** [comment]
There is no need to close this PR and create a new one for each version. You could have rebased this one and retargeted to 15.0.  The fw-bot will take care to propagate your changes to all versions up to master.

**Pull request closed without merging**